future-apps/payroll/README.md

```diff
@@ -1,71 +1,185 @@
-# [Payroll](https://github.com/aragon/aragon-apps/tree/master/future-apps/payroll)
+# Payroll
 
-_**Code in Github:**_ [aragon-apps/apps/payroll](https://github.com/aragon/aragon-apps/tree/master/future-apps/payroll)
+An automated employee payroll system featuring real-time salaries in multiple tokens.
 
-The purpose of the Payroll app is to implement a Payroll system in multiple currencies.
+## Capabilities
 
-### Initialization
+The Payroll app defines the following roles:
+
+- `ADD_EMPLOYEE_ROLE`: allows invocation of `addEmployee()`
+- `TERMINATE_EMPLOYEE_ROLE`: allows invocation of `terminateEmployee()`
+- `SET_EMPLOYEE_SALARY_ROLE`: allows invocation of `setEmployeeSalary()`
+- `ADD_BONUS_ROLE`: allows invocation of `addBonus()`
+- `ADD_REIMBURSEMENT_ROLE`: allows invocation of `addReimbursement()`
+- `ALLOWED_TOKENS_MANAGER_ROLE`: allows invocation of `addAllowedToken()`
+- `CHANGE_PRICE_FEED_ROLE`: allows invocation of `setPriceFeed()`
+- `MODIFY_RATE_EXPIRY_ROLE`: allows invocation of `setRateExpiryTime()`
+
+These roles can all be considered "management" roles, in that they are sensitive and meant to be held by more privileged members or apps in the organization. In particular, the set of `ADD_EMPLOYEE_ROLE`, `TERMINATE_EMPLOYEE_ROLE`, `SET_EMPLOYEE_SALARY_ROLE`, `ADD_BONUS_ROLE`, and `ADD_REIMBURSEMENT_ROLE` roles control functionality related to managing employees and their salaries, and the set of `ALLOWED_TOKENS_MANAGER_ROLE`, `CHANGE_PRICE_FEED_ROLE`, `MODIFY_RATE_EXPIRY_ROLE` roles control specific parameters and configuration points in how the Payroll app operates as a whole.
+
+Employees themselves are authenticated based on their "employee account" (an address), and have access to functionality related to their account that only they are allowed to access:
+
+- `changeAddressByEmployee()`
+- `determineAllocation()`
+- `payday()`
+
+### Payroll management
+
+The Payroll app is [initialized](#initialization) with four parameters:
+
+- **Finance**: a reference to a [Finance](https://github.com/aragon/aragon-apps/tree/master/apps/finance) instance that will be used to make payments
+  - Finance in turn will use a [Vault](https://github.com/aragon/aragon-apps/tree/master/apps/vault) to access funds, but having the Payroll app use a Finance app allows salary payments to record more obvious transactions in the organization's UI
+- **Denomination token**: ["base"](https://www.investopedia.com/terms/b/basecurrency.asp) token used to denominate all salaries in this Payroll instance
+  - This denomination token may not be a real on-chain currency (depends on how the linked price feed is implemented and if the denomination token is allowed to be used as a salary token)
+- **Price feed**: [`IFeed`](https://github.com/aragon/ppf/blob/master/packages/ppf-contracts/) implementation that provides exchange rates between different tokens
+  - May include exchange rates to/from "phantom" tokens, to mimic off-chain currencies (e.g. USD)
```

Comment:
- This price feed is only used to query the exchange rate of the allowed tokens based on the denomination token set.
```suggestion
  - May include exchange rates to/from "phantom" tokens, to mimic off-chain currencies (e.g. USD)
  - This price feed is only used to query the exchange rate of the allowed tokens based on the denomination token set.
```